src/main/scala/chiseltest/simulator/DebugPrintWrapper.scala

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+package chiseltest.simulator
+
+/** Wraps a [[SimulatorContext]] and prints the result of all operations performed on
+  * it to stdout
+  */
+class DebugPrintWrapper(inner: SimulatorContext) extends SimulatorContext {
+  private var cycle: Long = 0
+  override def sim = inner.sim
+  override def step(n: Int): Unit = {
+    val start = cycle
+    cycle += n
+    inner.step(n)
+    println(s"step $start -> $cycle")
+  }
+  override def peek(signal: String) = {
+    val res = inner.peek(signal)
+    println(s"$signal -> $res")
+    res
+  }
+  override def poke(signal: String, value: BigInt): Unit = {
+    inner.poke(signal, value)
+    println(s"$signal <- $value")
+  }
+  override def peekMemory(signal: String, index: Long) = {
+    val res = inner.peekMemory(signal, index)
+    println(s"$signal[$index] -> $res")
+    res
+  }
+
+  override def pokeMemory(signal: String, index: Long, value: BigInt): Unit = {
+    println(s"$signal[$index] <- $value")
+    inner.pokeMemory(signal, index, value)
```

Comment:
Do you want to be consistent in whether the print happens before or after the pokes / other actions?

Reply:
good catch